app/dbOperations/subjects.go

```diff
@@ -0,0 +1,172 @@
+)
+
+func GetSubjects(cosmos gremcos.Cosmos) ([]model.Subject, error) {
+	g := api.NewGraph("g")
+	query := g.V().HasLabel("subject")
+
+	res, err := cosmos.ExecuteQuery(query)
+
+	if err != nil {
+		fmt.Println("Failed to execute a gremling command")
+		//logger.Error().Err(err).Msg("Failed to execute a gremlin command")
+		return nil, err
+	}
+
+	response := api.ResponseArray(res)
+	vertices, err := response.ToVertices()
+
+	if err != nil {
+		return nil, err
+	}
+
+	subjects := verticesToSubjects(vertices)
+	return subjects, nil
+}
+
+func GetSubjectByName(cosmos gremcos.Cosmos, name string) (model.Subject, error) {
+	var subject model.Subject
+	g := api.NewGraph("g")
+	query := g.V().HasLabel("subject").Has("name", name)
+
+	res, err := cosmos.ExecuteQuery(query)
+	if err != nil {
+		fmt.Println("Failed to execute a gremling command")
+		//logger.Error().Err(err).Msg("Failed to execute a gremlin command")
+		return subject, err
+	}
+
+	return getSubjectFromResponse(res)
+}
+
+func InsertSubject(cosmos gremcos.Cosmos, subject model.Subject) (model.Subject, error) {
+	_, err := GetSubjectByName(cosmos, subject.Name)
+
+	if err == nil {
+		return model.Subject{}, errors.New("There is already a subject with this name")
+	}
+
+	g := api.NewGraph("g")
+
+	query := g.AddV("subject").Property("partitionKey", "subject")
+	query = addVertexProperties(query, subject)
+
+	res, err := cosmos.ExecuteQuery(query)
+	if err != nil {
+		fmt.Println("Failed to execute a gremling command")
+		//logger.Error().Err(err).Msg("Failed to execute a gremlin command")
+		return subject, err
+	}
+
+	return getSubjectFromResponse(res)
+}
+
+func UpdateSubject(cosmos gremcos.Cosmos, subject model.Subject, name string) (model.Subject, error) {
+	oldSubject, err := GetSubjectByName(cosmos, name)
+
+	fmt.Print("Old subject: ", oldSubject, "\n\n")
+	fmt.Print("Subject: ", subject, "\n\n\n")
+
+	if err != nil {
+		return model.Subject{}, errors.New("There is no subject with this name")
+	}
+
+	g := api.NewGraph("g")
+	query := addVertexProperties(g.VByStr(oldSubject.Id), subject)
+
+	res, err := cosmos.ExecuteQuery(query)
+	if err != nil {
+		fmt.Println("Failed to execute a gremling command")
+		//logger.Error().Err(err).Msg("Failed to execute a gremlin command")
+		return subject, err
+	}
+
+	return getSubjectFromResponse(res)
+}
+
+func DeleteSubject(cosmos gremcos.Cosmos, name string) error {
+	g := api.NewGraph("g")
+	query := g.V().HasLabel("subject").Has("name", name).Drop()
+
+	_, err := cosmos.ExecuteQuery(query)
+
+	return err
+}
+
+func addVertexProperties(vertex interfaces.Vertex, subject model.Subject) interfaces.Vertex {
+	vertex = vertex.
+		Property("name", subject.Name).
+		Property("difficulty", subject.Difficulty).
+		Property("category", subject.Category)
+
+	for i, r := range subject.References {
+		if i == 0 {
+			vertex = vertex.Property("references", r)
+		} else {
+			vertex = vertex.PropertyList("references", r)
+		}
+	}
```

Comment:
No need for this `if` to be inside this loop.
Something like the code below seems cleaner:

```
if len(subject.References) > 0 {
  vertex = vertex.Property("references", subject.References[0])
  for _, r := range subject.References[1..] {
    vertex = vertex.PropertyList("references", r)
  }
}
```